Library/Homebrew/sandbox.rb

```diff
@@ -89,6 +105,53 @@ def deny_write_homebrew_repository
     end
   end
 
+  def deny_read_shell_files
+    # Configuration.
+    # deny_read "/Users/#{ENV["USER"]}/.bash_profile"
+    # deny_read "/Users/#{ENV["USER"]}/.cshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.config/fish/config.fish"
+    # deny_read "/Users/#{ENV["USER"]}/.kshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.tcshrc"
+    # deny_read "/Users/#{ENV["USER"]}/.zshrc"
+
+    # Maybe this way is cleaner, although less explicit.
+    # Should this use a "/[^/]+/" regex instead of explicit USER to ensure
+    # we're not just protecting the active user on the machine but all of them?
+    %w[bash_profile cshrc config/fish/config.fish kshrc tcshrc zshrc].each do |s|
+      deny_read "/Users/#{ENV["USER"]}/.#{s}"
+    end
+
+    # Same as above. Seek review on preference.
+    # History.
+    # deny_read "/Users/#{ENV["USER"]}/.bash_history"
+    # deny_read "/Users/#{ENV["USER"]}/.zsh_history"
+    # deny_read "/Users/#{ENV["USER"]}/.config/fish/fish_history"
+    # deny_read "/Users/#{ENV["USER"]}/.sh_history"
+    # deny_read "/Users/#{ENV["USER"]}/.history"
+    %w[bash_history zsh_history config/fish/fish_history sh_history history].each do |s|
+      deny_read "/Users/#{ENV["USER"]}/.#{s}"
+    end
+  end
+
+  def deny_read_broad_home
```

Comment:
Yeah, difficult one to name. Essentially designed to block access to the vast majority of the directories and files under `$HOME` aside from certain exemptions. `broad_home` was working shorthand for `most_of_home`, which isn't ideal either really. If you're not fussed about deviating from the `deny_` and `allow_` language this could be something like `read_protect_user_exclusive_home_elements` or something. I think we're gonna fail for a short tile that is actually accurate & descriptive here though 😅.